io-engine/src/core/block_device.rs

```diff
@@ -82,6 +85,14 @@ pub trait BlockDevice {
         &self,
         listener: DeviceEventSink,
     ) -> Result<(), CoreError>;
+
+    fn is_zoned(&self) -> bool;
```

Comment:
Please add meaningful comments for all new methods, ideally with args and return value descriptions.